sugondat-shim/src/cmd/query/blob.rs

```diff
@@ -0,0 +1,50 @@
+use super::connect_rpc;
+use crate::cli::query::{blob::Params, BlockRef};
+
+use std::io::Write;
+
+pub async fn run(params: Params) -> anyhow::Result<()> {
+    let Params {
+        rpc,
+        block,
+        index,
+        raw,
+    } = params;
+
+    let client = connect_rpc(rpc).await?;
+
+    let maybe_hash = match block {
+        BlockRef::Best => None,
+        BlockRef::Hash(h) => Some(h),
+        BlockRef::Number(n) => Some(
+            client
+                .block_hash(n)
+                .await?
+                .ok_or_else(|| anyhow::anyhow!("No block with number {}", n))?
+                .0,
```

Comment:
goddammit. I was trying to guard the API from precisely this thing and failed!

Reply:
Specifically the `.0` thing? 

I did notice there's a comment in the `sugondat_rpc` file: "NOTE: we specifically avoid prolifiration of subxt types around the codebase. To that end, we avoid returning H256 and instead return [u8; 32] directly."

And yet, `pub async fn block_hash(&self, height: u64) -> anyhow::Result<Option<H256>>` returns the H256. Seems like an easy fix, though.

Reply:
Yes, that's precisely what I meant. If you look at blame, my name will pop on both of those lines.